docs/config/gson_config.md

```diff
@@ -0,0 +1,136 @@
+# Config files
+
+Sooner or later you'll get to the point where you'd like to make certain things toggleable.
+For such things, we can utilize included `Gson` library to read and write json configs.
+
+## Why gson?
+
+As mentioned above, `Gson` is already available in vanilla minecraft, meaning that you won't need
+any extra dependencies. If you don't need anything special and you're fine with `json` format, `Gson` will
+probably cover your needs. Not to mention that it will auto add the missing fields when updating
+the config without need to overwrite old values.
+
+## Config structure
+
+We'll use a custom config class that has fields that user / admin will be able to change.
+See below example.
+
+```java
+public class MyModConfig {
+
+    public float activationRange = 8.0F;
+
+    public boolean show = true;
+
+	public String message = "This is a config guide.";
+    
+}
+```
+We can access our config values like this:
+```java
+MyModConfig config = new MyModConfig();
+
+if(config.show)
+	System.out.println(config.message);
+
+config.show = false;
+
+// Do other stuff with values ...
+```
+
+We can present the config object in the following json structure.
+
+```json
+{
+	"activationRange": 8.0,
+	"show": true,
+	"message": "This is a config guide."
+}
+```
+
+## Saving, reading and parsing
+
+If `config` only exists in our memory, it's not that usefull, as changes will be reset
+after server restart. If we want to make them permanent, we should write the config to a file.
+
+```java
+public class MyModConfig {
```

Comment:
Could GSON handle this being a record?

Reply:
I'm sure gson (at least one bundled with Minecraft) errors with them

Reply:
Seems like that it's not implemented yet; https://github.com/google/gson/issues/1794.